testcontainers/src/core/ports.rs

```diff
@@ -105,6 +129,30 @@ impl TryFrom<PortMap> for Ports {
     }
 }
 
+impl Display for ExposedPort {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}/{}", self.port, self.protocol)
+    }
+}
+
+impl Display for InternetProtocol {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.encode())
+    }
+}
+
+impl From<u16> for ExposedPort {
+    fn from(port: u16) -> Self {
+        ExposedPort { port ,protocol: InternetProtocol::Tcp }
+    }
+}
+
+impl From<(u16, InternetProtocol)> for ExposedPort {
+    fn from((port, protocol): (u16, InternetProtocol)) -> Self {
+        ExposedPort { port, protocol }
+    }
+}
```

Comment:
Tbh, I'd prefer explicit API, instead of tuples and implicit conversion to TCP.
It's breaking change, but we anyway change `Image` api. So I think it's fine

Otherwise it's unclear when it was mapped to TCP or just consumed as is (e.g see my comment above for getting the port)

To clarify, I like idea of using explicit enum or functions, like:

`ExposedPort::Tcp(8080)` over `(8080, Protocol::Tcp)`

Implicit conversion also may produce inconsistent code, e.g:

```rs
let tcp_port = 8080:
let udp_port = (8080, Protocol::Udp)
```

Reply:
Done, also, I modfied PortMapping to be in sync with the new ExposedPort structure, it will be two breaking changes:

1. `GenericImage.with_exposed_port`
2. `Image.with_mapped_port`

Let me know if we need to maintain some backward compatibility in those two.

Reply:
> Let me know if we need to maintain some backward compatibility in those two.

Next release is going to contain several breaking changes. We can proceed with it, but better to provide a guide how to migrate to a new version

Reply:
Let me know where and I can help with the migration guide related to this PR

Reply:
Feel free to add this into PR description under separate section. I'll include this into release description too, collecting all changes. 
Something like:

### Migration guide
1. ...


Thank you!

Reply:
Added some documentation, let me know if that is enough

Reply:
More than enough, thank you!